hummingbot/connector/gateway/clob_spot/data_sources/gateway_clob_api_data_source_base.py

```diff
@@ -1,170 +1,495 @@
+        return resp["markets"]
+
+    async def _update_snapshots_loop(self):
+        while True:
+            update_task = safe_ensure_future(coro=self._emit_snapshots_updates())
+            min_delay_sleep = self._sleep(delay=self.min_snapshots_update_interval)
+            max_delay_sleep = self._sleep(delay=self.max_snapshots_update_interval)
+
+            or_group = next(
+                asyncio.as_completed(
+                    [update_task, max_delay_sleep]
+                )  # returns first done, lets the other run - i.e. run for at most max_delay_sleep
+            )
+            and_group = safe_gather(min_delay_sleep, or_group)  # run for at least min_delay_sleep
+
+            start_ts = self._time()
+            await and_group
+
+            if self._time() - start_ts >= self.max_snapshots_update_interval:
+                self.logger().warning(f"Snapshot update took longer than {self.max_snapshots_update_interval}.")
+
+    async def _emit_snapshots_updates(self):
+        tasks = [
+            self._emit_trading_pair_snapshot_update(trading_pair=trading_pair)
+            for trading_pair in self._trading_pairs
+        ]
+        await safe_gather(*tasks)
+
+    async def _emit_trading_pair_snapshot_update(self, trading_pair: str):
+        try:
+            snapshot_msg = await self.get_order_book_snapshot(trading_pair=trading_pair)
+            self._publisher.trigger_event(event_tag=OrderBookDataSourceEvent.SNAPSHOT_EVENT, message=snapshot_msg)
+        except asyncio.CancelledError:
+            raise
+        except Exception:
+            self.logger().error(f"Failed to update snapshot message for {trading_pair}.")
+
+    async def _get_ticker_data(self, trading_pair: str) -> List[Dict[str, Any]]:
+        ticker_data = await self._get_gateway_instance().get_clob_ticker(
+            connector=self.connector_name,
+            chain=self._chain,
+            network=self._network,
+            trading_pair=trading_pair,
+        )
+        return ticker_data["markets"]
+
+    async def _on_create_order_transaction_failure(self, order: GatewayInFlightOrder, order_result: Dict[str, Any]):
+        raise ValueError(
+            f"The creation transaction for {order.client_order_id} failed. Please ensure you have sufficient"
+            f" funds to cover the transaction gas costs."
+        )
+
+    async def _on_cancel_order_transaction_failure(self, order: GatewayInFlightOrder, cancelation_result: Dict[str, Any]):
+        raise ValueError(
+            f"The cancelation transaction for {order.client_order_id} failed. Please ensure you have sufficient"
+            f" funds to cover the transaction gas costs."
+        )
+
+    async def _wait_for_transaction_to_be_included_in_a_block_and_request_order_status_update(
+        self, in_flight_order: GatewayInFlightOrder
+    ):
+        await self._sleep(delay=self.current_block_time)
+        await self.get_order_status_update(in_flight_order=in_flight_order)
```

Comment:
TBH, I don't think we should do this.
We can just keep polling for the status of the transaction in the status polling loop.
Inflight orders can then be updated when the transaction is processed.

This will cause unnecessary increment of the not-found counter especially during times the network is congested.

Reply:
Agreed. I was on the fence about this inclusion. Removing it.